src/lib.rs

```diff
@@ -92,252 +110,27 @@
+pub use crate::error::{ErrorReason, GuardError, Error};
+pub use crate::pod::PodTransmutable;
 
-/// Trasform a byte vector into a vector of an arbitrary type.
-///
-/// The vector's allocated byte buffer will be reused when possible, and
-/// have as many instances of a type as will fit, rounded down.
-/// Extraneous data is ignored.
-///
-/// # Examples
-///
-/// ```
-/// # use safe_transmute::guarded_transmute_vec_permissive;
-/// # include!("../tests/test_util/le_to_native.rs");
-/// # fn main() {
-/// // Little-endian
-/// # unsafe {
-/// # /*
-/// assert_eq!(guarded_transmute_vec_permissive::<u16>(vec![0x00, 0x01, 0x00, 0x02]),
-/// # */
-/// # assert_eq!(guarded_transmute_vec_permissive::<u16>(vec![0x00, 0x01, 0x00, 0x02].le_to_native::<u16>()),
-///            vec![0x0100, 0x0200]);
-/// # /*
-/// assert_eq!(guarded_transmute_vec_permissive::<u32>(vec![0x04, 0x00, 0x00, 0x00, 0xED]),
-/// # */
-/// # assert_eq!(guarded_transmute_vec_permissive::<u32>(vec![0x04, 0x00, 0x00, 0x00, 0xED].le_to_native::<u32>()),
-///            vec![0x0000_0004]);
-/// assert_eq!(guarded_transmute_vec_permissive::<u16>(vec![0xED]), vec![]);
-/// # }
-/// # }
-/// ```
+pub use crate::to_bytes::{safe_transmute_to_bytes, safe_transmute_one_to_bytes, guarded_transmute_to_bytes_pod_many, guarded_transmute_to_bytes_pod};
 #[cfg(feature = "std")]
-pub unsafe fn guarded_transmute_vec_permissive<T>(mut bytes: Vec<u8>) -> Vec<T> {
-    PermissiveGuard::check::<T>(&bytes).unwrap();
-    let ptr = bytes.as_mut_ptr();
-    let capacity = bytes.capacity() / size_of::<T>();
-    let len = bytes.len() / size_of::<T>();
-    forget(bytes);
-    Vec::from_raw_parts(ptr as *mut T, len, capacity)
-}
+pub use crate::to_bytes::{safe_transmute_to_bytes_vec, guarded_transmute_to_bytes_pod_vec};
 
-/// Trasform a byte vector into a vector of an arbitrary type.
-///
-/// The vector's allocated byte buffer will be reused when possible, and
-/// should not have extraneous data.
-///
-/// # Examples
-///
-/// ```
-/// # use safe_transmute::guarded_transmute_vec_pedantic;
-/// # include!("../tests/test_util/le_to_native.rs");
-/// # fn main() {
-/// // Little-endian
-/// # unsafe {
-/// # /*
-/// assert_eq!(guarded_transmute_vec_pedantic::<u16>(vec![0x00, 0x01, 0x00, 0x02])?,
-/// # */
-/// # assert_eq!(guarded_transmute_vec_pedantic::<u16>(vec![0x00, 0x01, 0x00, 0x02].le_to_native::<u16>()).unwrap(),
-///            vec![0x0100, 0x0200]);
-///
-/// assert!(guarded_transmute_vec_pedantic::<u32>(vec![0x04, 0x00, 0x00, 0x00, 0xED]).is_err());
-/// # }
-/// # }
-/// ```
 #[cfg(feature = "std")]
-pub unsafe fn guarded_transmute_vec_pedantic<T>(bytes: Vec<u8>) -> Result<Vec<T>, Error> {
-    PedanticGuard::check::<T>(&bytes)?;
-    Ok(guarded_transmute_vec_permissive(bytes))
-}
+pub use crate::bool::{safe_transmute_bool_vec_permissive, safe_transmute_bool_vec_pedantic, guarded_transmute_bool_vec_permissive, guarded_transmute_bool_vec_pedantic};
```

Comment:
A decent sprinkling of `#[allow(deprecated)]` around here (maybe on separated `use` blocks?) would be nice.